src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupsMap: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const keyValue: string = String(item[key]);
+
+    if (!groupsMap[keyValue]) {
+      groupsMap[keyValue] = [item];
+    } else {
+      groupsMap[keyValue].push(item);
+    }
```

Comment:
```suggestion
    if (!groupsMap[keyValue]) {
      groupsMap[keyValue] = [];
    } 
      groupsMap[keyValue].push(item);
```

We can avoid using else block